Symfony/CS/Fixer/PSR1/Psr4Fixer.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\PSR1;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\StdinFileInfo;
+use Symfony\CS\Tokenizer\Token;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author Jordi Boggiano <j.boggiano@seld.be>
+ * @author Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ * @author Bram Gotink <bram@gotink.me>
+ * @author Graham Campbell <graham@alt-three.com>
+ */
+final class Psr4Fixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAnyTokenKindsFound(Token::getClassyTokenKinds());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isRisky()
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $namespace = false;
+        $classyName = null;
+        $classyIndex = 0;
+
+        foreach ($tokens as $index => $token) {
+            if ($token->isGivenKind(T_NAMESPACE)) {
+                if (false !== $namespace) {
+                    return;
+                }
+
+                $namespace = true;
+            } elseif ($token->isClassy()) {
+                if (null !== $classyName) {
+                    return;
+                }
+
+                $classyIndex = $tokens->getNextNonWhitespace($index);
```

Comment:
That is broken in the psr0 fixer too since I basically started off by copying that class.